src/naive/datetime/mod.rs

```diff
@@ -129,6 +155,36 @@ impl NaiveDateTime {
         datetime.expect("invalid or out-of-range datetime")
     }
 
+    /// Creates a new [NaiveDateTime] from milliseconds or microseconds since the UNIX epoch.
```

Comment:
Style nit: I would prefer this (private) API to be ordered below the public users of this method.